service/tablet_allocator.cc

```diff
@@ -1123,7 +1123,10 @@ class load_balancer {
 
             int min_rack_load = std::numeric_limits<int>::min();
             int max_rack_load = std::numeric_limits<int>::max();
-            for (const auto& [_, load] : rack_load) {
+            for (const auto& [rack, load] : rack_load) {
+                if (rack == src_info.rack()) {
+                    continue;
+                }
```

Comment:
It seems to change the logic, we ignore the source rack in the calculation. Where before we detected rack imbalance, now we may not.